lib/psci/psci_off.c

```diff
@@ -83,6 +83,12 @@ int psci_do_cpu_off(unsigned int end_pwrlvl)
 	psci_stats_update_pwr_down(end_pwrlvl, &state_info);
 #endif
 
+	/*
+	 * Plat. management: Perform platform specific actions to turn this
+	 * cpu off e.g. exit cpu coherency, program the power controller etc.
+	 */
+	psci_plat_pm_ops->pwr_domain_off(&state_info);
```

Comment:
Moving this handler above the `cpu power down` ops will break for platforms which have dependancy on this (eg: CCI programming). So we cannot do this change generically.

Introducing an `early` pwr_domain_off/pwr_domain_suspend hook is an option but that requires further discussion. On a similar note, we could have `late` pwr_domain_in_finisher/pwr_domain_suspend_finisher hook for operations that needs to be done after `psci_do_pwrup_cache_maintenance()`.